ArduCopter/mode_systemid.cpp

```diff
@@ -115,65 +154,76 @@ void ModeSystemId::exit()
 // should be called at 100hz or more
 void ModeSystemId::run()
 {
-    // apply simple mode transform to pilot inputs
-    update_simple_mode();
-
-    // convert pilot input to lean angles
     float target_roll, target_pitch;
-    get_pilot_desired_lean_angles(target_roll, target_pitch, copter.aparm.angle_max, copter.aparm.angle_max);
-
-    // get pilot's desired yaw rate
-    float target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->norm_input_dz());
-
-    if (!motors->armed()) {
-        // Motors should be Stopped
-        motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::SHUT_DOWN);
-    // Tradheli doesn't set spool state to ground idle when throttle stick is zero.  Ground idle only set when
-    // motor interlock is disabled.
-    } else if (copter.ap.throttle_zero && !copter.is_tradheli()) {
-        // Attempting to Land
-        motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::GROUND_IDLE);
-    } else {
-        motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
-    }
+    float target_yaw_rate = 0.0f;
+    float pilot_throttle_scaled = 0.0f;
+    float target_climb_rate = 0.0f;
+    Vector2f input_vel;
+
+    if ((AxisType)axis.get() != AxisType::DISTURB_POS_LAT && (AxisType)axis.get() != AxisType::DISTURB_POS_LONG 
+         && (AxisType)axis.get() != AxisType::DISTURB_VEL_LAT && (AxisType)axis.get() != AxisType::DISTURB_VEL_LONG
+         && (AxisType)axis.get() != AxisType::INPUT_LOITER_LAT && (AxisType)axis.get() != AxisType::INPUT_LOITER_LONG) {
```

Comment:
Could this whole section be put in to a new function? So we would have `ModeSystemId::run_pos_control()` and `ModeSystemId::run_angle_rate_mixer_control()` (I'm sure there is a better name we could come up with.) Then this core `run` function would just call into one or the other based on `axis`.

Reply:
Some of this happens prior to the SID_AXIS case statement because the targets are determined for the stabilize flight mode from the pilot's inputs.  Not sure I want to mess with this.  I could make the if statement conditional look cleaner

Reply:
I cleaned up the conditional statement to make it look cleaner and read easier.  I'm don't think it is wise to break this out as @IamPete1 suggests.